internal/batch/service.go

```diff
@@ -0,0 +1,112 @@
+package batch
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	goabatch "github.com/artefactual-labs/enduro/internal/api/gen/batch"
+	"github.com/artefactual-labs/enduro/internal/cadence"
+	"github.com/artefactual-labs/enduro/internal/collection"
+	"github.com/artefactual-labs/enduro/internal/validation"
+	"github.com/go-logr/logr"
+	"go.uber.org/cadence/.gen/go/shared"
+	cadenceclient "go.uber.org/cadence/client"
+)
+
+var ErrBatchStatusUnavailable = errors.New("batch status unavailable")
+
+//go:generate mockgen  -destination=./fake/mock_batch.go -package=fake github.com/artefactual-labs/enduro/internal/batch Service
+
+type Service interface {
+	Submit(context.Context, *goabatch.SubmitPayload) (res *goabatch.BatchResult, err error)
+	Status(context.Context) (res *goabatch.BatchStatusResult, err error)
+	InitProcessingWorkflow(ctx context.Context, batchDir, key, pipelineName string) error
+}
+
+type batchImpl struct {
+	logger logr.Logger
+	cc     cadenceclient.Client
+}
+
+var _ Service = (*batchImpl)(nil)
+
+func NewService(logger logr.Logger, cc cadenceclient.Client) *batchImpl {
+	return &batchImpl{
+		logger: logger,
+		cc:     cc,
+	}
+}
+
+func (s *batchImpl) Submit(ctx context.Context, payload *goabatch.SubmitPayload) (*goabatch.BatchResult, error) {
+	var pipelineName string
+	if payload.Pipeline != nil {
+		pipelineName = *payload.Pipeline
+	}
```

Comment:
I can't remember if we decided to leave this out, but it would be nice if an empty pipeline name or undefined attribute results in picking up one automatically from the registry.

Reply:
Yes, **I** decided to leave it out for now. I wonder if "when empty pick the first from the registry" can be problematic in a multi-pipeline setup.